LandSandBoat/server
# [core] [sql] [lua] [SMN] Implement Siren BP Rage, implement Magic Evasion boost/Magic Evasion Down, various fixes

**_I affirm:_**

## What does this pull request do?

Implements all Siren's rage BPs
Implements Magic Evasion Down/Magic Evasion Boost
Various fixes found along the way:

- CLuaBaseEntity::getBaseDelay() needing to round up the delay due to truncation
- CLuaBaseEntity::getBaseDelay() was adding delay when it didn't need to for H2H
- Add new TP multipliers upon discovering that pets apparently gained 3x TP arbitrarily and killed that "feature" too
- change CLuaBaseEntity::setTP(N) to actually _set_ tp to N instead of getting potential side effects from calling addTP later
- Avatar Favor bloodpact delay implemented

## Steps to test these changes

Use Siren BPs, get 95 TP return if physical, 0 TP if magical.
Verify absorb of Hysteric Assault (acts like Sanguine where it adds HP equal to damage)
Verify dispel of Sonic Buffet
